streams/src/test/java/org/apache/kafka/streams/integration/PauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,422 @@
+        assertFalse(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY1));
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY2));
+
+        awaitOutput(OUTPUT_STREAM_1, 10, COUNT_OUTPUT_DATA_ALL);
+        assertTopicSize(OUTPUT_STREAM_1, 10);
+        assertTopicSize(OUTPUT_STREAM_2, 5);
+    }
+
+    @Test
+    public void pauseResumeShouldWorkAcrossInstances() throws Exception {
+        produceToInputTopics(INPUT_STREAM_1, STANDARD_INPUT_DATA);
+
+        kafkaStreams = buildKafkaStreams(OUTPUT_STREAM_1);
+        kafkaStreams.pause();
+        kafkaStreams.start();
+
+        waitForApplicationState(singletonList(kafkaStreams), State.RUNNING, STARTUP_TIMEOUT);
+        assertTrue(kafkaStreams.isPaused());
+
+        kafkaStreams2 = buildKafkaStreams(OUTPUT_STREAM_2);
+        kafkaStreams2.pause();
+        kafkaStreams2.start();
+        waitForApplicationState(singletonList(kafkaStreams2), State.RUNNING, STARTUP_TIMEOUT);
+        assertTrue(kafkaStreams2.isPaused());
+
+        waitUntilStreamsHasPolled(kafkaStreams, 2);
+        waitUntilStreamsHasPolled(kafkaStreams2, 2);
+        assertTopicSize(OUTPUT_STREAM_1, 0);
+
+        kafkaStreams2.close();
+        kafkaStreams2.cleanUp();
+        waitForApplicationState(singletonList(kafkaStreams2), State.NOT_RUNNING, STARTUP_TIMEOUT);
+
+        kafkaStreams.resume();
+        waitForApplicationState(singletonList(kafkaStreams), State.RUNNING, STARTUP_TIMEOUT);
+
+        awaitOutput(OUTPUT_STREAM_1, 5, COUNT_OUTPUT_DATA);
+    }
+
+    @Test
+    public void pausedTopologyShouldNotRestoreStateStores() throws Exception {
+        final Properties properties1 = props();
+        properties1.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1);
+        final Properties properties2 = props();
+        properties2.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1);
+        produceToInputTopics(INPUT_STREAM_1, STANDARD_INPUT_DATA);
+
+        kafkaStreams = buildKafkaStreams(OUTPUT_STREAM_1, properties1);
+        kafkaStreams2 = buildKafkaStreams(OUTPUT_STREAM_1, properties2);
+        kafkaStreams.start();
+        kafkaStreams2.start();
+
+        waitForApplicationState(Arrays.asList(kafkaStreams, kafkaStreams2), State.RUNNING, STARTUP_TIMEOUT);
+
+        awaitOutput(OUTPUT_STREAM_1, 5, COUNT_OUTPUT_DATA);
+
+        kafkaStreams.close();
+        kafkaStreams2.close();
+
+        kafkaStreams = buildKafkaStreams(OUTPUT_STREAM_1, properties1);
+        kafkaStreams2 = buildKafkaStreams(OUTPUT_STREAM_1, properties2);
+        kafkaStreams.cleanUp();
+        kafkaStreams2.cleanUp();
+
+        kafkaStreams.pause();
+        kafkaStreams2.pause();
+        kafkaStreams.start();
+        kafkaStreams2.start();
+
+        waitForApplicationState(Arrays.asList(kafkaStreams, kafkaStreams2), State.REBALANCING, STARTUP_TIMEOUT);
+
+        assertStreamsLagStaysConstant(kafkaStreams);
+        assertStreamsLagStaysConstant(kafkaStreams2);
```

Comment:
nit:  `assertStreamsLagStaysConstant()` -> `assertStreamsLocalStoreLagStaysConstant()`

Reply:
Changed it!